src/protocols/secure_channel/PASESession.cpp

```diff
@@ -825,6 +825,9 @@ CHIP_ERROR PASESession::OnMessageReceived(ExchangeContext * exchange, const Payl
     MsgType msgType = static_cast<MsgType>(payloadHeader.GetMessageType());
     SuccessOrExit(err);
 
+    // Once we receive a message for the PASE protocol, we know the Peer is active
+    mExchangeCtxt->SetPeerActiveStateHint(true);
```

Comment:
My basic claim is that (1) this is true for all protocols and (2) we can therefore handle it entirely inside the exchange.

Reply:
True, but that would require the exchange context checking the message type when receiving a message which is not something we should add in the exchange context imo. It should be up to the caller to determine if it thinks the peer (receiver) is active or not.

Reply:
Why does it require checking the message type?

Reply:
As it stands, from with in the exchange context, we do not check what is the message being sent. To `SetPeerActiveStateHint` in the exchange context, it would require the ExchangeContext checking if we are sending sigma3 for example or making sure we are not sending the first message of PASE.

Reply:
>  it would require the ExchangeContext checking if we are sending sigma3 for example or making sure we are not sending the first message of PAS

Why?  Why wouldn't we just check "this is not the first message being sent on this exchange", or perhaps "we have received a message on this exchange before"?